**Configuracoes**


In [2]:
library(ggplot2)

In [3]:
library(dplyr)


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




In [4]:
packageurl <- "https://cran.r-project.org/src/contrib/Archive/DescTools/DescTools_0.99.30.tar.gz"
install.packages(packageurl, repos=NULL, type="source")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Warning message in install.packages(packageurl, repos = NULL, type = "source"):
“installation of package ‘/tmp/RtmpYSKBMG/downloaded_packages/DescTools_0.99.30.tar.gz’ had non-zero exit status”


In [5]:
library(DescTools)

ERROR: Error in library(DescTools): there is no package called ‘DescTools’


In [ ]:
options(repr.plot.width = 7, repr.plot.height = 4)

**Importando dataset do projeto**

In [9]:
dados <- read.csv("../input/dados-ibge-domicilios-2015/dados.csv")

In [ ]:
head(dados, 5)

In [ ]:
nrow(dados)

# **Teste de normalidade**

**Nível de significância (α)**

In [ ]:
significancia <- 0.05 

**Testando a variável renda**

In [ ]:
ggplot(dados, aes(x = Renda)) +
    geom_histogram() + 
    ylab("Frequência") +
    xlab("Renda") + 
    ggtitle("Histograma Renda") +
    theme(plot.title = element_text(hjust = 0.5))

**Critério do valor p**

In [ ]:
set.seed(2011)
amostra <- sample_n(dados, 5000)

In [ ]:
resultado <- shapiro.test(amostra$Renda)

resultado

In [ ]:
p_valor <- resultado$p.value

p_valor

In [ ]:
resultado$statistic

In [ ]:
p_valor <= significancia

**Testeando a variável altura**

In [ ]:
ggplot(dados, aes(x = Altura)) +
    geom_histogram() + 
    ylab("Frequência") +
    xlab("Altura") + 
    ggtitle("Histograma Altura") +
    theme(plot.title = element_text(hjust = 0.5))

In [ ]:
resultado <- shapiro.test(amostra$Altura)

resultado

In [ ]:
p_valor <- resultado$p.value

p_valor

In [ ]:
resultado$statistic

In [ ]:
p_valor <= significancia

# **Teste bicaudal**

# A empresa Suco Bom produz sucos de frutas em embalagens de 500 ml. Seu processo de produção é quase todo automatizado e as embalagens de sucos são preenchidas por uma máquina que às vezes apresenta um certo desajuste, levando a erros no preenchimento das embalagens para mais ou menos conteúdo. Quando o volume médio cai abaixo de 500 ml, a empresa se preocupa em perder vendas e ter problemas com os orgãos fiscalizadores. Quando o volume passa de 500 ml, a empresa começa a se preocupar com prejuízos no processo de produção.

# O setor de controle de qualidade da empresa Suco Bom extrai, periodicamente, amostras de 50 embalagens para monitorar o processo de produção. Para cada amostra, é realizado um teste de hipóteses para avaliar se o maquinário se desajustou. A equipe de controle de qualidade assume um nível de significância de 5%.

# Suponha agora que uma amostra de 50 embalagens foi selecionada e que a média amostral observada foi de 503,24 ml. Esse valor de média amostral é suficientemente maior que 500 ml para nos fazer rejeitar a hipótese de que a média do processo é de 500 ml ao nível de significância de 5%?**

In [ ]:
amostra <- c(509, 505, 495, 510, 496, 509, 497, 502, 503, 505, 
           501, 505, 510, 505, 504, 497, 506, 506, 508, 505, 
           497, 504, 500, 498, 506, 496, 508, 497, 503, 501, 
           503, 506, 499, 498, 509, 507, 503, 499, 509, 495, 
           502, 505, 504, 509, 508, 501, 505, 497, 508, 507)

In [ ]:
amostra <- data.frame(Amostra = amostra)

head(amostra)

In [ ]:
media_amostra <- mean(amostra$Amostra)

media_amostra

In [ ]:
desvio_padrao_amostra <- sd(amostra$Amostra)

desvio_padrao_amostra

In [ ]:
media <- 500

In [ ]:
significancia <- 0.05

In [ ]:
confianca <- 1 - significancia

In [ ]:
n <- nrow(amostra)

n

**Passo 1 - formulação das hipóteses**

*H0 : média = 500*

*H1 : média != 500*

**Passo 2 - escolha da distruição amostral adequada**

*N > 30. Sigma desconhecido. Então utilizo distribuição normal junto com o desvio padrao da amostra*

**Passo 3 - fixação da significância do teste (alfa)**

In [ ]:
probabilidade <- 0.5 + confianca / 2 

probabilidade

In [ ]:
z_alpha_2 <- qnorm(probabilidade)

z_alpha_2

**Passo 4 - Cálculo da estátistica-teste e verificação desse valor com as áreas de aceitação e rejeição do teste**

In [ ]:
z <- (media_amostra - media) / (desvio_padrao_amostra / sqrt(n))

z

**Passo 5 - Aceitação ou rejeição da hipótese nula**

In [ ]:
z <= - z_alpha_2

In [ ]:
z >= z_alpha_2

In [ ]:
if(z <= - z_alpha_2 || z >= z_alpha_2){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

**Solução utilizando p valor**

In [ ]:
p_valor <- 2 * (1 - pnorm(z))

p_valor

In [ ]:
if(p_valor <= significancia){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

**Utilizando DescTools**

In [ ]:
resultado <- ZTest(amostra$Amostra, mu = media, sd_pop = desvio_padrao_amostra)

In [ ]:
resultado$statistic

In [ ]:
z

In [ ]:
p_valor <- resultado$p.value

p_valor

In [ ]:
if(p_valor <= significancia){
    
    "Rejeitar H0"
    
} else{
    
    "Aceitar H0"
    
}

# Uma indústria de cimento afirma que a quantidade média de cimento nas embalagens de seu produto é de 25 kg. Um teste de pesagem em uma amostra de 45 embalagens apresentou um peso médio igual a 25,2 kg. Medições anteriores afirmam que a distribuição dos pesos segue uma normal e que o desvio padrão populacional é igual a 400 g. Considerando um nível de significância igual a 5%, responda as seguintes questões:

**1) Qual a hipótese nula a ser testada?**

**2) Qual o valor da estatística de teste?**

**3) Qual a conclusão do teste?**

**Passo 1 - formulação das hipóteses**

*H0 : média = 25 kg*

*H1 : média != 25 kg*

**Passo 2 - escolha da distribuição amostral adequada**

*N > 30. Desvio padrao populacional é conhecido. Então utilizo distribuição normal e desvio padrao populacional*

**Passo 3 - fixação da significância do teste**

In [ ]:
probabilidade <- 0.5 + 0.95 / 2 

probabilidade

In [ ]:
z_alpha_2 <- qnorm(probabilidade)

z_alpha_2

**Passo 4 - Cálculo da estátistica-teste e verificação desse valor com as áreas de aceitação e rejeição do teste**

In [ ]:
z <- (25.2 - 25) / (0.4/ sqrt(45))

z

**Passo 5 - Aceitação ou rejeição da hipótese nula**

In [ ]:
if(z <= - z_alpha_2 || z >= z_alpha_2){
    
    "Rejeitar H0"
    
} else{
    
    "Aceitar H0"
    
}

**1**

A hipótese nula a ser testada é "A quantidade média de cimento nas embalagens da empresa em questão é de 25kg"

**2**

In [ ]:
z

**3**

In [ ]:
if(z <= - z_alpha_2 || z >= z_alpha_2){
    
    "Rejeitar H0"
    
} else{
    
    "Aceitar H0"
    
}

# Considere novamente os dados do problema anterior. Uma indústria de cimento afirma que a quantidade média de cimento nas embalagens de seu produto é de 25 kg. Um teste de pesagem em uma amostra de 45 embalagens apresentou um peso médio igual a 25,2 kg. Medições anteriores afirmam que a distribuição dos pesos segue uma normal e que o desvio padrão populacional é igual a 400 g.

**Com um nível de significância igual a 5%, teste a hipótese nula de que a média do processo é igual a 25 kg e selecione a alternativa que indica o p-valor do teste aplicado (considere somente 5 casas decimais).**

In [ ]:
p_valor <- 2 * (1 - pnorm(z))

p_valor

In [ ]:
if(p_valor <= significancia){
    
    "Rejeitar H0"
    
} else{
    
    "Aceitar H0"
    
}

# Considere o seguinte teste de hipóteses:
# 
# H0 : média = 22
# 
# H1 : média != 22
# 
# Uma amostra aleatória de tamanho 75 foi selecionada e se sabe que o desvio padrão da população é 10. Obtenha o p-valor e a sua conclusão para os dados amostrais abaixo:
# 
# 1) Média amostral igual a 25
# 
# 2) Média amostral igual a 20
# 
# Considere um nível de significância de 1%.

In [ ]:
media <- 22

In [ ]:
n <- 75

In [ ]:
sd <- 10

In [ ]:
significancia <- 0.01

In [ ]:
confianca <- 1 - significancia

confianca

In [ ]:
probabilidade <- 0.5 + confianca / 2

probabilidade

In [ ]:
z_alpha_2 <- qnorm(probabilidade)

z_alpha_2

In [ ]:
calcular_p_valor <- function(media_amostra, media, desvio_padrao_amostra, n){
    
    z <- (media_amostra - media) / (desvio_padrao_amostra / sqrt(n))

    p_valor <- 2 * (1 - pnorm(abs(z)))
    
    return(p_valor)
    
}

In [ ]:
for(i in c(20, 25)){
    
    p_valor <- calcular_p_valor(i, media, sd, n)
    
    print(sprintf("Para a média amostral de %d, o valor de p é %f", i, p_valor))
    
}

In [ ]:
p_valor <- 2 * (1 - pnorm(abs(z)))

p_valor

# **Teste unicaudal**

# Um famoso fabricante de refrigerantes alega que uma lata de 350 ml de seu principal produto contém, no máximo, 37 gramas de açúcar. Esta alegação nos leva a entender que a quantidade média de açúcar em uma lata de refrigerante deve ser igual ou menor que 37 g.

# Um consumidor desconfiado e com conhecimentos em inferência estatística resolve testar a alegação do fabricante e seleciona, aleatóriamente, em um conjunto de estabelecimentos distintos, uma amostra de 25 latas do refrigerante em questão. Utilizando o equipamento correto o consumidor obteve as quantidades de açúcar em todas as 25 latas de sua amostra.

# Assumindo que essa população se distribua aproximadamente como uma normal e considerando um nível de significância de 5%, é possível aceitar como válida a alegação do fabricante?

In [ ]:
q <- seq(0.05, 0.005, by = -0.005)
df <- seq(1, 30, by = 1)
probabilidade <- c()
for(i in df){
    for(j in q){
        probabilidade <- c(probabilidade, qt(j, i, lower.tail = F))
    }
}
tabela_t_student <- matrix(probabilidade, ncol=10, byrow=TRUE)
colnames(tabela_t_student) <- format(q)
rownames(tabela_t_student) <- format(df)
tabela_t_student

In [ ]:
amostra <- c(37.27, 36.42, 34.84, 34.60, 37.49, 
           36.53, 35.49, 36.90, 34.52, 37.30, 
           34.99, 36.55, 36.29, 36.06, 37.42, 
           34.47, 36.70, 35.86, 36.80, 36.92, 
           37.04, 36.39, 37.32, 36.64, 35.45)

In [ ]:
head(amostra)

In [ ]:
media_amostra <- mean(amostra)

media_amostra

In [ ]:
desvio_padrao_amostra <- sd(amostra)

desvio_padrao_amostra

In [ ]:
media <- 37

In [ ]:
significancia <- 0.05

In [ ]:
confianca <- 1 - significancia

confianca

In [ ]:
n <- 25

In [ ]:
graus_de_liberdade <- n - 1

graus_de_liberdade

**Passo 1 - formulação das hipóteses**

*H0 : média <= 37*

*H1 : média > 37*

**Passo 2 - escolha da distribuição amostral adequada**

*N < 30. Se distribui aproxidamente como uma normal. Desvio padrão populacional desconhecido. Então utilizo distribuição de student e desvio padrao amostral*

**Passo 3 - fixação da significancia do teste (alfa)**

In [ ]:
tabela_t_student[23:25, ]

In [ ]:
t_alpha <- qt(confianca, graus_de_liberdade)

t_alpha

**Passo 4 - Cálculo da estátistica-teste e verificação desse valor com as áreas de aceitação e rejeição do teste**

In [ ]:
t <- (media_amostra - media) / (desvio_padrao_amostra / sqrt(n))

t

**Passo 5 - Aceitação ou rejeição da hipótese nula**

In [ ]:
if(t >= t_alpha){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

**Usando t.test**

In [ ]:
resultado <- t.test(amostra, alternative = "greater", mu = media)

resultado

In [ ]:
resultado$statistic

In [ ]:
p_valor <- resultado$p.value

In [ ]:
p_valor <= significancia

# **A empresa Limpa Esgoto garante ser capaz de realizar o tratamento de esgoto e obter, no máximo, 150g de impurezas para cada mil litros de esgoto tratado. Vinte amostras de mil litros de esgoto apresentaram, em média, 230g de impurezas e desvio padrão amostral igual a 90 g. Assumindo alfa igual a 5% e população normalmente distribuída, seria possível discordar da empresa Limpa Esgoto?**

In [ ]:
media <- 150

In [ ]:
n <- 20

In [ ]:
graus_de_liberdade <- n - 1

graus_de_liberdade

In [ ]:
media_amostra <- 230

In [ ]:
sd_amostra <- 90

In [ ]:
significancia <- 0.05 

In [ ]:
confianca <- 1 - significancia

confianca

In [ ]:
t_alpha <- qt(confianca, graus_de_liberdade)

t_alpha

In [ ]:
t <- (media_amostra - media) / (sd_amostra / sqrt(n))

t

In [ ]:
if(t >= t_alpha){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# **A pizzaria Muito Queijo alega que a quantidade de queijo em suas pizzas tamanho família é de, no mínimo, 350g. Uma amostra de 35 pizzas tamanho família revelou uma média de 330 g de queijo por pizza. O desvio padrão amostral foi de 80g. Assumindo alfa igual a 5% e população normalmente distribuída, seria possível discordar da alegação da pizzaria?**

In [ ]:
media <- 350

In [ ]:
n <- 35

In [ ]:
media_amostra <- 330

In [ ]:
sd_amostra <- 80

In [ ]:
significancia <- 0.05

In [ ]:
confianca <- 1 - significancia

confianca

In [ ]:
z_alpha <- qnorm(1 - significancia)

z_alpha

In [ ]:
z <- (media_amostra - media) / (sd_amostra / sqrt(n))

z

In [ ]:
if(z <= -z_alpha){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# Considere o seguinte teste de hipóteses:
# 
# H0 : média >= 75
# 
# H1 : média < 75
# 
# Uma amostra aleatória de tamanho 23 foi selecionada e se sabe que o desvio padrão da amostra é 12. Obtenha o p-valor e a sua conclusão para os dados amostrais abaixo:
# 
# 1) Média amostral igual a 82
# 
# 2) Média amostral igual a 70
# 
# Considere um nível de confiança de 95%.

In [ ]:
media <- 75

In [ ]:
n <- 23

In [ ]:
graus_de_liberdade <- 23 - 1

graus_de_liberdade

In [ ]:
sd_amostra <- 12

In [ ]:
confianca <- 0.95

In [ ]:
significancia <- 1 - confianca

significancia

In [ ]:
t <- (82 - media) / (sd_amostra / sqrt(n))

t

In [ ]:
p_valor <- pt(t, graus_de_liberdade, lower.tail = T)

p_valor

In [ ]:
calcular_p_valor <- function(media_amostra, media, sd_amostra, n){
    
    t <- (media_amostra - media) / (sd_amostra / sqrt(n))
    
    p_valor <- pt(t, n - 1, lower.tail = T)
    
    return(p_valor)
    
}

In [ ]:
for(i in c(82, 70)){
    
    p_valor <- calcular_p_valor(i, media, sd_amostra, n)
    
    print(sprintf("Para a média amostral de %d, o valor de p é %.2f", i, p_valor))
    
    if(p_valor <= significancia){
        
        print("Rejeitar H0")
        
    } else{
        
        print("Aceitar H0")
        
    }
    
}

# **Uma lanchonete afirma que suas vendas médias são de exatamente reais 32,00. Uma amostra aleatória de 20 vendas foi selecionada e os valores são apresentados no dataset em anexo**
# **Sabe-se que as vendas são normalmente distribuídas. Obtenha o valor da estatística de teste, o p-valor e a decisão para o teste. Assuma um nível de significância de 5%.**

In [ ]:
vendas <- c(32.45, 26.8, 27.81, 30.22, 30.88, 24.9, 31.94, 16.02, 
    24.39, 26.01, 21.83, 25.35, 22.46, 38.2, 29.86, 22.79, 28.83, 27.34, 32.22, 33.26)

In [ ]:
dataset <- data.frame(Amostra = vendas)

In [ ]:
media <- 32

In [ ]:
significancia <- 0.05

In [ ]:
resultado <- t.test(dataset, alternative = "two.sided", mu = media)

resultado

# Testes para duas amostras

# Em nosso dataset temos os rendimento dos chefes de domicílio obtidos da Pesquisa Nacional por Amostra de Domicílios - PNAD no ano de 2015. Um problema bastante conhecido em nosso país diz respeito a desigualdade de renda, principalmente entre homens e mulheres.

# Duas amostras aleatórias, uma de 500 homens e outra com 500 mulheres, foram selecionadas em nosso dataset. Com o objetivo de comprovar tal desigualdade, teste a igualdade das médias entre estas duas amostras com um nível de significância de 1%.

In [ ]:
homens <- matrix(c(300, 1000, 4000, 5000, 300, 350, 1200, 1200, 1100, 1800, 700, 5000, 250, 1560, 400, 5500, 3000, 0, 1100, 1500, 500, 500, 788, 500, 3000, 788, 2400, 788, 3300, 1350, 2500, 2000, 1300, 2430, 9000, 10120, 1380, 2000, 1400, 1000, 500, 1500, 2200, 2100, 4000, 1000, 1200, 2900, 1800, 2000, 788, 1576, 800, 2400, 788, 788, 788, 0, 12000, 7880, 850, 1000, 1000, 1600, 3800, 788, 1200, 300, 350, 700, 6000, 1500, 5000, 2000, 1200, 800, 250, 800, 1600, 400, 3000, 2304, 800, 1400, 450, 788, 2200, 6000, 1200, 4000, 6000, 1100, 1200, 1300, 3000, 3000, 1500, 1280, 788, 1400, 788, 4000, 0, 1000, 2500, 1300, 3000, 500, 1600, 2000, 1280, 150, 0, 0, 7000, 1061, 700, 788, 2000, 788, 10500, 788, 1600, 1050, 1100, 3000, 800, 1400, 1200, 2500, 1000, 1500, 1500, 900, 2000, 300, 3000, 788, 1400, 2000, 2000, 3000, 2600, 15000, 1500, 950, 1200, 1500, 7500, 1400, 350, 750, 1200, 788, 0, 2500, 1700, 788, 700, 1600, 1200, 320, 6000, 2000, 2000, 3000, 900, 2100, 0, 788, 1800, 1600, 4000, 1300, 1300, 1500, 8900, 1400, 788, 600, 1000, 950, 2000, 4000, 2300, 2000, 800, 2500, 1500, 1400, 800, 6000, 788, 900, 200, 1300, 788, 2600, 1500, 8000, 900, 2000, 2000, 350, 3000, 9580, 0, 400, 800, 1700, 1800, 1200, 18000, 9000, 3000, 1200, 700, 1200, 400, 987, 6000, 2000, 0, 0, 480, 500, 800, 1680, 10000, 1200, 1700, 788, 2200, 10000, 10000, 100, 1200, 1600, 2500, 1300, 1200, 1300, 2200, 200, 60, 1100, 1200, 6000, 4500, 100, 788, 2900, 2500, 900, 788, 2500, 4000, 788, 1400, 1000, 300, 788, 1000, 4000, 2200, 788, 1000, 1000, 600, 1600, 6000, 2500, 2500, 1000, 3000, 2200, 4500, 1500, 4300, 1500, 8500, 3200, 1200, 1200, 1500, 4000, 2000, 1350, 1500, 890, 5000, 2400, 13000, 800, 1500, 1500, 2000, 580, 500, 920, 1200, 2000, 788, 500, 20000, 1350, 1576, 500, 8000, 1800, 600, 1000, 3000, 4000, 1500, 788, 600, 2300, 1500, 500, 500, 800, 900, 8000, 1600, 3000, 788, 1500, 5200, 900, 3565, 650, 1700, 1600, 788, 1200, 788, 788, 100, 900, 1700, 10000, 1600, 200, 788, 1400, 1500, 400, 4200, 1400, 4000, 5516, 3500, 700, 1400, 1200, 0, 7000, 2000, 5000, 2000, 750, 3000, 2000, 1500, 200, 3000, 1700, 1500, 7000, 1500, 788, 2400, 2500, 1500, 2000, 1500, 2000, 1100, 800, 1800, 480, 3500, 700, 3400, 5000, 5000, 1300, 3000, 2000, 900, 2000, 1000, 1800, 0, 1500, 788, 3500, 875, 5000, 2000, 1300, 1600, 0, 750, 1200, 120, 1800, 1200, 788, 1000, 0, 1400, 2300, 7000, 4000, 5000, 788, 20000, 4000, 0, 1200, 1800, 500, 1000, 788, 1300, 5000, 50, 1700, 2000, 1600, 2000, 1350, 1500, 600, 1700, 780, 2446, 2100, 5000, 1700, 1200, 1000, 788, 4500, 1500, 788, 0, 1580, 1000, 4500, 2400, 788, 1100, 2000, 788, 100, 1200, 1200, 1200, 1000, 2000, 788, 2000, 15000, 600, 0, 1500, 3000, 4000, 900, 810, 600, 1500, 4000, 1200, 5000, 5300, 2500, 800, 0, 1400, 1500, 4000, 1200, 400, 1000, 820, 1000, 1000, 788, 1500, 2500, 1500, 220, 600, 788, 1750, 7000))
mulheres <- matrix(c(788, 1200, 6000, 1000, 788, 1100, 1900, 1577, 900, 950, 1200, 788, 788, 1100, 30, 620, 900, 1000, 1200, 2000, 0, 500, 1200, 1500, 1200, 1120, 788, 788, 2300, 2400, 3000, 788, 4000, 1000, 500, 500, 1700, 200, 6000, 400, 950, 1100, 50, 930, 850, 1100, 3500, 1500, 1200, 900, 1100, 1500, 788, 1000, 788, 4500, 4000, 8000, 3500, 788, 1050, 1000, 1400, 3500, 600, 3000, 500, 930, 2660, 788, 360, 2364, 788, 160, 1100, 1085, 1050, 1500, 100, 0, 788, 250, 1700, 1300, 800, 0, 2000, 820, 910, 300, 2000, 200, 788, 788, 800, 1500, 1300, 1200, 0, 600, 1036, 400, 1100, 788, 400, 1500, 1200, 1800, 1000, 788, 850, 60, 1800, 3500, 600, 200, 500, 200, 1100, 1540, 1100, 900, 800, 500, 200, 1200, 1250, 788, 500, 200, 788, 200, 0, 1800, 2000, 1000, 900, 3000, 700, 1200, 788, 2800, 3300, 400, 0, 850, 1022, 6000, 750, 1000, 3500, 400, 1500, 1000, 800, 0, 980, 2400, 850, 1100, 788, 1100, 788, 1200, 788, 864, 1000, 500, 400, 3000, 1200, 100, 80, 900, 2000, 1250, 1000, 300, 400, 1500, 60, 8000, 1000, 600, 800, 350, 788, 0, 600, 788, 2500, 1300, 800, 8000, 1100, 800, 900, 2000, 0, 800, 1400, 1000, 1200, 788, 3840, 788, 3940, 788, 560, 800, 2000, 600, 2900, 0, 400, 800, 7000, 788, 788, 788, 788, 1250, 1500, 1386, 100, 300, 788, 788, 600, 600, 900, 2800, 788, 350, 900, 0, 150, 788, 788, 1000, 30, 788, 780, 900, 0, 0, 3000, 1800, 2000, 1100, 788, 0, 7500, 1800, 788, 788, 2000, 3000, 180, 2500, 800, 0, 1250, 200, 1100, 0, 1100, 400, 1300, 300, 0, 1312, 600, 200, 4000, 2400, 12000, 2150, 500, 400, 0, 600, 400, 788, 1200, 500, 2000, 1500, 70, 500, 2000, 1000, 100, 500, 2000, 100, 2115, 800, 5800, 2300, 750, 788, 0, 900, 1000, 600, 3000, 5500, 1600, 788, 4000, 0, 1100, 120, 320, 1100, 2500, 200, 800, 3000, 550, 4200, 7000, 720, 0, 2000, 5000, 2000, 788, 500, 788, 7000, 500, 788, 1760, 1200, 3500, 1500, 150, 800, 788, 3000, 400, 788, 1000, 2000, 2500, 0, 788, 5200, 788, 0, 600, 1300, 120, 850, 1576, 788, 2000, 1300, 788, 800, 1800, 655, 1580, 789, 788, 850, 788, 7000, 788, 430, 800, 788, 900, 10000, 1200, 300, 400, 788, 788, 3000, 900, 1300, 1300, 788, 800, 1000, 1890, 0, 788, 900, 4000, 788, 1010, 150, 450, 1700, 890, 1200, 2800, 1000, 788, 788, 700, 600, 800, 880, 6000, 800, 800, 820, 788, 1030, 560, 2100, 6500, 2500, 788, 400, 300, 788, 2300, 1000, 3500, 984, 1576, 420, 1700, 450, 1800, 400, 1500, 1500, 4256, 1200, 890, 1200, 300, 400, 850, 1500, 3800, 1800, 996, 2000, 1000, 788, 1500, 300, 600, 950, 1200, 788, 1200, 1500, 250, 788, 1200, 6000, 300, 789, 1500, 788, 3800, 780, 1200, 1200, 220, 788, 500, 200, 480, 1576, 1576, 1035, 900, 800, 1000, 805, 200, 1200, 2220, 1500, 880, 220, 2000, 788, 150, 0, 13000, 40, 5500, 788, 788, 1000, 400, 2000, 200, 1600))

In [ ]:
media_amostra_M <- mean(mulheres)

media_amostra_M

In [ ]:
desvio_padrao_amostra_M <- sd(mulheres)

desvio_padrao_amostra_M

In [ ]:
media_amostra_H <- mean(homens)

media_amostra_H

In [ ]:
desvio_padrao_amostra_H <- sd(homens)

desvio_padrao_amostra_H

In [ ]:
significancia <- 0.01

confianca <- 1 - significancia

n_M <- length(mulheres)

n_H <- length(homens)

D_0 <- 0

**Passo 1 - formulação das hipóteses**

*u1: média das rendas dos chefes dedomicílios do sexo masculino*

*u2: média das rendas dos chefes de domicílios do sexo feminino*

*H0 : u1 <= u2*

*H1 : u1 > u2*

*ou*

*H0: u1 - u2 <= 0*

*H1: u1 - u2 > 0*

**Passo 2 - escolha das distribuição amostral adequada**

*N > 30. Desvio padrao populacional desconhecido. Utilizo desvio padrao amostral*

**Passo 3 - fixação da significância do teste (alfa)**

In [ ]:
probabilidade <- confianca

In [ ]:
z_alpha <- qnorm(probabilidade)

z_alpha

**Passo 4 - Cálculo da estátistica-teste e verificação desse valor com as áreas de aceitação e rejeição do teste**

In [ ]:
numerador <- (media_amostra_H - media_amostra_M) - D_0

numerador

In [ ]:
denominador <- sqrt((desvio_padrao_amostra_H ** 2 / n_H) + (desvio_padrao_amostra_M ** 2 / n_M))

denominador

In [ ]:
z <- numerador / denominador

z

**Passo 5 - Aceitação ou rejeição da hipótese nula**

In [ ]:
if(z >= z_alpha){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

**utilizando p valor**

In [ ]:
p_valor <- pnorm(z, lower.tail = F)

p_valor

In [ ]:
if(p_valor <= significancia){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

In [ ]:
install.packages("BSDA")

In [ ]:
library(BSDA)

**z.test**

In [ ]:
resultado <- z.test(x = homens,
                   y = mulheres,
                   alternative = 'greater',
                   mu = 0,
                   sigma.x = sd(homens),
                   sigma.y = sd(mulheres),
                   conf.level = 0.99)

resultado

In [ ]:
p_valor <- resultado$p.value

p_valor

In [ ]:
if(p_valor <= significancia){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

**t.test**

In [ ]:
resultado <- t.test(x = homens,
                   y = mulheres,
                   alternative = 'greater',
                   paired = F,
                   var.equal = T)

resultado

In [ ]:
p_valor <- resultado$p.value 

p_valor

In [ ]:
if(p_valor <= significancia){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# Uma prova de estatística foi aplicada em duas salas de aula diferentes. Na sala A, tínhamos 50 alunos e a nota média foi de 5,3, com desvio padrão de 2,1. Na sala B eram 55 alunos, que alcançaram nota média de 7,6 pontos, com desvio padrão de 2,8. Considerando um nível de significância de 5% e que a distribuição das notas segue uma normal, é possível afirmar que as notas médias dessas duas salas foram diferentes?



# *h0: uA <= uB*

# *h1: uA > uB*

In [ ]:
n_a <- 50

In [ ]:
media_a <- 5.3

In [ ]:
sd_a <- 2.1

In [ ]:
n_b <- 55

In [ ]:
media_b <- 7.6

In [ ]:
sd_b <- 2.8

In [ ]:
significancia <- 0.05

In [ ]:
confianca <- 1 - significancia

In [ ]:
probabilidade <- confianca

In [ ]:
z_alpha <- qnorm(probabilidade)

z_alpha

In [ ]:
D_0 = 0

In [ ]:
numerador <- media_a - media_b - D_0

numerador

In [ ]:
denominador <- sqrt((sd_a  ** 2 / n_a) + (sd_b ** 2 / n_b))

denominador

In [ ]:
z <- numerador / denominador

z

In [ ]:
if(z <= -z_alpha_2 || z >= z_alpha_2){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# Uma imobiliária da cidade do Rio de Janeiro alega que o valor do m² no bairro do Leblon está avaliado em, no máximo, reais 1000,00 a mais, quando comparado ao bairro de Ipanema. Uma pesquisa realizada com uma amostra de 13 imóveis de cada bairro revelou os seguintes parâmetros:

# Leblon: Média (reais por m2) = 21800 e Desvio Padrão (reais por m2) = 450
# Ipanema: Média (reais por m2) = 21300 e Desvio Padrão (reais por m2) = 320

# É possível concordar com a afirmação da imobiliária? Assuma um nível de significância de 5% e populações normalmente distribuídas

In [ ]:
D_0 <- 1000

In [ ]:
n <- 13

In [ ]:
media_L <- 21800

In [ ]:
sd_L <- 450

In [ ]:
media_I <- 20300

In [ ]:
sd_I <- 320

In [ ]:
significancia <- 0.05

In [ ]:
confianca <- 1 - significancia

confianca

In [ ]:
graus_liberdade <- 2 * n - 2

graus_liberdade

In [ ]:
t_alpha <- qt(confianca, graus_liberdade)

t_alpha

In [ ]:
numerador <- (media_L - media_I) - D_0

numerador

In [ ]:
denominador <- sqrt((sd_L ** 2 / n) + (sd_I ** 2 / n))

denominador

In [ ]:
t <- numerador / denominador

t

In [ ]:
if(z <= -z_alpha_2 || z >= z_alpha_2){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# Testes não paramétricos

# **Teste Qui-quadrado**

# Antes de cada partida do campeonato nacional de futebol, as moedas utilizadas pelos árbitros devem ser verificadas para se ter certeza de que não são viciadas, ou seja, que não tendam para determinado resultado. Para isso um teste simples deve ser realizado antes de cada partida. Este teste consiste em lançar a moeda do jogo 50 vezes e contar as frequências de CARAS e COROAS obtidas. A tabela abaixo mostra o resultado obtido no experimento:

       Cara | Coroa
Observado 17 | 33

Esperado 25  | 25

# A um nível de significância de 5%, é possível afirmar que a moeda não é honesta, isto é, que a moeda apresenta uma probabilidade maior de cair com a face CARA voltada para cima?

In [1]:
p <- c(0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 0.75, 0.9, 0.975, 0.95, 0.99, 0.995)
df <- seq(1, 30, by = 1)
probabilidade <- c()
for(i in df){
    for(j in p){
        probabilidade <- c(probabilidade, round(qchisq(j, i, lower.tail = T), 4))
    }
}
tabela_chi_2 <- matrix(probabilidade, ncol=13, byrow=TRUE)
colnames(tabela_chi_2) <- format(p)
rownames(tabela_chi_2) <- format(df)
tabela_chi_2

0.005   0.010   0.025   0.050   0.100   0.250   0.500   0.750   0.900  
 1  0.0000  0.0002  0.0010  0.0039  0.0158  0.1015  0.4549  1.3233  2.7055
 2  0.0100  0.0201  0.0506  0.1026  0.2107  0.5754  1.3863  2.7726  4.6052
 3  0.0717  0.1148  0.2158  0.3518  0.5844  1.2125  2.3660  4.1083  6.2514
 4  0.2070  0.2971  0.4844  0.7107  1.0636  1.9226  3.3567  5.3853  7.7794
 5  0.4117  0.5543  0.8312  1.1455  1.6103  2.6746  4.3515  6.6257  9.2364
 6  0.6757  0.8721  1.2373  1.6354  2.2041  3.4546  5.3481  7.8408 10.6446
 7  0.9893  1.2390  1.6899  2.1673  2.8331  4.2549  6.3458  9.0371 12.0170
 8  1.3444  1.6465  2.1797  2.7326  3.4895  5.0706  7.3441 10.2189 13.3616
 9  1.7349  2.0879  2.7004  3.3251  4.1682  5.8988  8.3428 11.3888 14.6837
10  2.1559  2.5582  3.2470  3.9403  4.8652  6.7372  9.3418 12.5489 15.9872
11  2.6032  3.0535  3.8157  4.5748  5.5778  7.5841 10.3410 13.7007 17.2750
12  3.0738  3.5706  4.4038  5.2260  6.3038  8.4384 11.3403 14.8454 18.5493
13  3.5650  4.1069  5.0088  5.8919  7.0415  9.2991 12.3398 15.9839 19.8119
14  4.0747  4.6604  5.6287  6.5706  7.7895 10.1653 13.3393 17.1169 21.0641
15  4.6009  5.2293  6.2621  7.2609  8.5468 11.0365 14.3389 18.2451 22.3071
16  5.1422  5.8122  6.9077  7.9616  9.3122 11.9122 15.3385 19.3689 23.5418
17  5.6972  6.4078  7.5642  8.6718 10.0852 12.7919 16.3382 20.4887 24.7690
18  6.2648  7.0149  8.2307  9.3905 10.8649 13.6753 17.3379 21.6049 25.9894
19  6.8440  7.6327  8.9065 10.1170 11.6509 14.5620 18.3377 22.7178 27.2036
20  7.4338  8.2604  9.5908 10.8508 12.4426 15.4518 19.3374 23.8277 28.4120
21  8.0337  8.8972 10.2829 11.5913 13.2396 16.3444 20.3372 24.9348 29.6151
22  8.6427  9.5425 10.9823 12.3380 14.0415 17.2396 21.3370 26.0393 30.8133
23  9.2604 10.1957 11.6886 13.0905 14.8480 18.1373 22.3369 27.1413 32.0069
24  9.8862 10.8564 12.4012 13.8484 15.6587 19.0373 23.3367 28.2412 33.1962
25 10.5197 11.5240 13.1197 14.6114 16.4734 19.9393 24.3366 29.3389 34.3816
26 11.1602 12.1981 13.8439 15.3792 17.2919 20.8434 25.3365 30.4346 35.5632
27 11.8076 12.8785 14.5734 16.1514 18.1139 21.7494 26.3363 31.5284 36.7412
28 12.4613 13.5647 15.3079 16.9279 18.9392 22.6572 27.3362 32.6205 37.9159
29 13.1211 14.2565 16.0471 17.7084 19.7677 23.5666 28.3361 33.7109 39.0875
30 13.7867 14.9535 16.7908 18.4927 20.5992 24.4776 29.3360 34.7997 40.2560
   0.975   0.950   0.990   0.995  
 1  5.0239  3.8415  6.6349  7.8794
 2  7.3778  5.9915  9.2103 10.5966
 3  9.3484  7.8147 11.3449 12.8382
 4 11.1433  9.4877 13.2767 14.8603
 5 12.8325 11.0705 15.0863 16.7496
 6 14.4494 12.5916 16.8119 18.5476
 7 16.0128 14.0671 18.4753 20.2777
 8 17.5345 15.5073 20.0902 21.9550
 9 19.0228 16.9190 21.6660 23.5894
10 20.4832 18.3070 23.2093 25.1882
11 21.9200 19.6751 24.7250 26.7568
12 23.3367 21.0261 26.2170 28.2995
13 24.7356 22.3620 27.6882 29.8195
14 26.1189 23.6848 29.1412 31.3193
15 27.4884 24.9958 30.5779 32.8013
16 28.8454 26.2962 31.9999 34.2672
17 30.1910 27.5871 33.4087 35.7185
18 31.5264 28.8693 34.8053 37.1565
19 32.8523 30.1435 36.1909 38.5823
20 34.1696 31.4104 37.5662 39.9968
21 35.4789 32.6706 38.9322 41.4011
22 36.7807 33.9244 40.2894 42.7957
23 38.0756 35.1725 41.6384 44.1813
24 39.3641 36.4150 42.9798 45.5585
25 40.6465 37.6525 44.3141 46.9279
26 41.9232 38.8851 45.6417 48.2899
27 43.1945 40.1133 46.9629 49.6449
28 44.4608 41.3371 48.2782 50.9934
29 45.7223 42.5570 49.5879 52.3356
30 46.9792 43.7730 50.8922 53.6720

In [ ]:
F_Observada <- c(17, 33)

In [ ]:
F_Esperada <- c(25, 25)

In [ ]:
significancia <- 0.05

In [ ]:
confianca <- 1 - significancia

In [ ]:
k <- 2 #Numero de eventos possiveis. binomial

In [ ]:
graus_de_liberdade <- k - 1

**Passo 1 - formulação das hipóteses**

*H0: Fcara = Fcoroa*

*H1: Fcara != Fcoroa*

**Passo 2 - fixação da significância do teste (alfa)**

In [ ]:
tabela_chi_2[1:3, ]

In [ ]:
chi_2_alpha <- qchisq(confianca, graus_de_liberdade)

chi_2_alpha

**Passo 3 - cálculo da estatística-teste e verificação desse valor com as áreas de aceitação e rejeição do teste**

In [ ]:
chi_2 <- ((F_Observada[1] - F_Esperada[1]) ** 2 / F_Esperada[1]) + ((F_Observada[2] - F_Esperada[2]) ** 2 / F_Esperada[2])

chi_2

**Passo 4 - Aceitação ou rejeição da hipótese nula**

In [ ]:
if(chi_2_alpha < chi_2){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

In [ ]:
chi_2 = 0

for(i in 1:k){
    
    chi_2 <- chi_2 + ((F_Observada[i] - F_Esperada[i]) ** 2 / F_Esperada[i])
    
}

chi_2

In [ ]:
if(chi_2_alpha < chi_2){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

In [ ]:
p_valor <- pchisq(chi_2, df = graus_de_liberdade, lower.tail = F)

p_valor

In [ ]:
if(chi_2_alpha < chi_2){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

**chisq.test**

In [ ]:
resultado <- chisq.test(x = F_Observada, p = c(0.5, 0.5))

resultado

In [ ]:
resultado$statistic

In [ ]:
resultado$parameter

In [ ]:
p_valor <- resultado$p.value

p_valor

In [ ]:
resultado$observed

In [ ]:
resultado$expected

In [ ]:
if(chi_2_alpha < chi_2){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# O resultado de 60 lançamentos de duas moedas apresentou a seguinte frequência de resultados:
# 
# Resultados | Frequências Observadas
# 2 CARAS    | 12
# 1 CARA E 1 COROA | 34
# 2 COROAS   | 14
# Total      | 60

In [ ]:
n <- 60

In [ ]:
significancia <- 0.05

In [ ]:
confianca <- 1 - significancia

confianca

In [ ]:
esperado <- c(15, 30, 15)

In [ ]:
observado <- c(12, 34, 14)

In [ ]:
k <- 3

In [ ]:
graus_de_liberdade <- 3 - 1

graus_de_liberdade

In [ ]:
chi2_alpha <- qchisq(confianca, graus_de_liberdade)

chi2_alpha

In [ ]:
chi_2 = 0

for(i in 1:k){
    
    chi_2 <- chi_2 + ((observado[i] - esperado[i]) ** 2 / esperado[i])
    
}

chi_2

# Na realização de um teste não paramétrico Qui-quadrado, com apenas 5 graus de liberdade, obteve-se a estatística de teste no valor de 7,45. Qual seria o p-valor para este teste?

In [ ]:
graus_de_liberdade <- 5

In [ ]:
chi_2 <- 7.45

In [ ]:
pchisq(chi_2, graus_de_liberdade, lower.tail = F)

# Considerando o problema que resolvemos anteriormente, onde o resultado de 60 lançamentos de duas moedas apresentou a seguinte frequência de resultados:

# Resultados | Frequências Observadas
# 2 CARAS    | 12
# 1 CARA E 1 COROA | 34
# 2 COROAS   | 14
# Total      | 60

# Considerando um nível de significância de 5%, seria possível dizer que as moedas estão bem equilibradas, ou seja, que apresentam probabilidades iguais de se obter CARA ou COROA?

In [ ]:
observado <- c(12, 34, 14)

In [ ]:
esperado <- c(0.25, 0.5, 0.25)

In [ ]:
resultado <- chisq.test(x = observado, p = esperado)

resultado

# **Teste Wilcoxon**

# Um novo tratamento para acabar com o hábito de fumar está sendo empregado em um grupo de 35 pacientes voluntários. De cada paciente testado foram obtidas as informações de quantidades de cigarros consumidos por dia antes e depois do término do tratamento. Assumindo um nível de confiança de 95% é possível concluir que, depois da aplicação do novo tratamento, houve uma mudança no hábito de fumar do grupo de pacientes testado?

In [ ]:
fumo <- data.frame(
    Antes = c(39, 25, 24, 50, 13, 52, 21, 29, 10, 22, 50, 15, 36, 39, 52, 48, 24, 15, 40, 41, 17, 12, 21, 49, 14, 55, 46, 22, 28, 23, 37, 17, 31, 49, 49),
    Depois = c(16, 8, 12, 0, 14, 16, 13, 12, 19, 17, 17, 2, 15, 10, 20, 13, 0, 4, 16, 18, 16, 16, 9, 9, 18, 4, 17, 0, 11, 14, 0, 19, 2, 9, 6)
)

In [ ]:
significancia <- 0.05

In [ ]:
confianca <- 1 - significancia

confianca

In [ ]:
n <- 35

In [ ]:
head(fumo)

In [ ]:
media_antes <- mean(fumo$Antes)

media_antes

In [ ]:
media_depois <- mean(fumo$Depois)

media_depois

**Passo 1 - formulação das hipóteses**

*H0 : média_antes = média_depois*

*H1 : média_antes > média_depois*

**Passo 2 - escolha das distribuição amostral adequada**

*N > 30. Utilizo a normal. (Lembrando que nã é um teste paramétrico)*

**Passo 3 - fixação da significância do teste (alfa)**

In [ ]:
probabilidade <- 0.5 + confianca / 2

probabilidade

In [ ]:
z_alpha_2 <- qnorm(probabilidade)

z_alpha_2

**Passo 4 - Cálculo da estátistica-teste e verificação desse valor com as áreas de aceitação e rejeição do teste**

**Construindo a tabela com os postos**

In [ ]:
fumo

In [ ]:
fumo['Dif'] <- fumo$Depois - fumo$Antes

head(fumo)

In [ ]:
fumo['|Dif|'] = abs(fumo$Dif)
fumo

In [ ]:
fumo <- fumo[order(fumo$'|Dif|'),]

In [ ]:
head(fumo, 7)

In [ ]:
fumo['Posto'] <- seq(1, nrow(fumo))

fumo

In [ ]:
posto <- aggregate(x = fumo$'Posto', by = list(fumo$'|Dif|'), FUN = mean)

In [ ]:
colnames(posto) <- c('|Dif|', 'Posto')
posto

In [ ]:
fumo$Posto <- NULL

head(fumo)

In [ ]:
fumo <- merge(x = fumo, y = posto, by = '|Dif|', all.x = T)

In [ ]:
head(fumo)

In [ ]:
fumo['Posto (+)'] <- apply(fumo[, c('Dif', 'Posto')], 1, function(x) if(x[1] > 0) x[2] else 0)
fumo

In [ ]:
fumo['Posto (-)'] <- apply(fumo[, c('Dif', 'Posto')], 1, function(x) if(x[1] < 0) x[2] else 0)
fumo

In [ ]:
fumo$Posto = NULL

In [ ]:
fumo

**Obtendo T**

**T = menor das somas de postos de mesmo sinal**

In [ ]:
T <- min( sum( fumo['Posto (+)'] ), sum( fumo['Posto (-)'] ) )
T

**Obtendo médiaT**

In [ ]:
mu_T <- (n * (n + 1)) / 4

mu_T

**Obtendo sigmaT**

In [ ]:
sigma_T <- sqrt( ( n * ( n + 1 ) * ( ( 2 * n ) + 1 ) ) / 24 )
sigma_T

**Obtendo Zteste**

In [ ]:
Z <- ( T - mu_T ) / sigma_T
Z

**Passo 5 - Aceitação ou rejeição da hipótese nula**

In [ ]:
if (Z <= - z_alpha || Z >= z_alpha){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

**Utilizando wilcox.test**

In [ ]:
resultado <- wilcox.test(fumo$Depois, fumo$Antes, paired = T, correct = F)

resultado

In [ ]:
p_valor <- resultado$p.value

p_valor

In [ ]:
if (p_valor <= significancia){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

### <font color='green'>Conclusão: Rejeitamos a hipótese de que não existe diferença entre os grupos, isto é, existe uma diferença entre as médias de cigarros fumados pelos pacientes antes e depois do tratamento. E como é possível verificar através das médias de cigarros fumados por dia antes (31.86) e depois (11.2) do tratamento, podemos concluir que o tratamento apresentou resultado satisfatório.</font>

# Desconfiado da eficiência dos cursos e materiais de estudo on-line da Alura, um professor resolveu realizar um teste com um grupo de 30 alunos de sua classe. Para isso, ele submeteu esses alunos a duas etapas distintas e logo depois de cada etapa, aplicou uma avaliação. Na etapa inicial, foram oferecidas aulas normais, sem a utilização do material de apoio da Alura. Na segunda etapa, também foram oferecidas aulas normais, mas com a utilização do material de apoio da Alura. As notas obtidas pelos alunos nas duas etapas estão no data frame abaixo:

In [ ]:
alunos <- data.frame(
    Sem_Alura = c(4, 3, 5, 4, 8, 6, 5, 7, 6, 7, 8, 7, 3, 7, 4, 7, 5, 7, 5, 7, 6, 4, 3, 3, 6, 3, 7, 4, 6, 6),
    Com_Alura = c(8, 6, 10, 3, 9, 5, 8, 5, 4, 10, 5, 9, 8, 5, 10, 8, 7, 9, 9, 5, 5, 7, 8, 9, 8, 5, 5, 8, 5, 9)
)

# Apresente o resultado do teste não paramétrico de Wilcoxon aplicado pelo professor (menor das somas de postos de mesmo sinal T, estatística de teste Z e decisão do teste). Considere um nível de significância de 10%.

In [ ]:
significancia <- 0.1

In [ ]:
confianca <- 1 - significancia

confianca

In [ ]:
probabilidade <- 0.5 + confianca / 2 

probabilidade

In [ ]:
n <- nrow(alunos)

n

In [ ]:
media_antes <- mean(alunos$Sem_Alura)

media_antes

In [ ]:
media_depois <- mean(alunos$Com_Alura)

media_depois

In [ ]:
z_alpha_2 <- qnorm(probabilidade)

z_alpha_2

In [ ]:
alunos['Dif'] <- alunos$Sem_Alura - alunos$Com_Alura

head(alunos)

In [ ]:
alunos['|Dif|'] <- abs(alunos$Dif)

head(alunos)

In [ ]:
alunos <- alunos[order(alunos['|Dif|']),]

head(alunos)

In [ ]:
alunos['Posto'] <- seq(1, nrow(alunos))

head(alunos)

In [ ]:
posto <- aggregate(x = alunos$Posto, by = list(alunos$'|Dif|'), FUN = mean)

In [ ]:
colnames(posto) <- c('|Dif|', 'Posto')

In [ ]:
posto

In [ ]:
alunos$Posto <- NULL

head(alunos)

In [ ]:
alunos <- merge(x = alunos, y = posto, by = '|Dif|', all.x = T)

head(alunos)

In [ ]:
alunos['Posto (+)'] <- apply(alunos[, c('Dif', 'Posto')], 1, function(x) if(x[1] > 0) x[2] else 0)
 
head(alunos)

In [ ]:
alunos['Posto (-)'] <- apply(alunos[, c('Dif', 'Posto')], 1, function(x) if(x[1] < 0) x[2] else 0)
    
head(alunos)    

In [ ]:
alunos$Posto <- NULL

head(alunos)

**Obtendo t**

In [ ]:
T <- min(sum( alunos['Posto (+)'] ), sum( alunos['Posto (-)'] ))

T

**Obtendo médiaT**

In [ ]:
mu_T <- n * (n + 1) / 4

mu_T

**Obtendo sigmaT**

In [ ]:
sigma_T <- sqrt( ( n * ( n + 1 ) * ( ( 2 * n ) + 1 ) ) / 24 )

sigma_T

**Obtendo Zteste**

In [ ]:
Z <- (T - mu_T) / sigma_T

Z

In [ ]:
if (Z <= - z_alpha || Z >= z_alpha){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# Uma fábrica de rações para porcos gostaria de verificar se a adição de um determinado composto químico em um de seus produtos teria um efeito significativo no ganho de peso dos animais. Para isso, selecionou uma amostra de 35 porcos e testou este mesmo grupo em duas situações distintas. No primeiro teste, alimentou os suínos exclusivamente com a ração sem o composto por uma semana e registrou a variação de peso de cada animal do grupo. No segundo teste, alimentou o mesmo grupo de animais com a ração com o novo composto, por mais uma semana, e registrou novamente a variação dos pesos.

# Os dados de variação de peso de cada suíno testado estão no data frame abaixo (em gramas):

In [ ]:
pesos <- data.frame(
    Sem_Composto = c(180, -39, 325, 303, 127, 149, 271, 163, 287, 255, 398, 324, 335, 421, 
        216, 274, 373, 116, 197, -37, 321, 431, 112, 304, 417, 362, 51, 187, 195, 304, 202, 158, 105, 90, 245),
    Com_Composto = c(484, 187, 442, 108, 488, 283, 286, 419, 240, 266, 198, 130, 484, 424, 
        145, 133, 282, 386, 408, 290, 429, 386, 318, 390, 347, 442, 440, 356, 517, 454, 401, 108, 228, 471, 495)
)

# Apresente o p-valor e a decisão correta do teste de Wilcoxon sobre a diferença entre as médias sem e com o composto na ração. Utilize a função wilcox.test() do R e considere um nível de significância de 5%.

In [ ]:
resultado <- wilcox.test(pesos$Com_Composto, pesos$Sem_Composto, pair = T, correct = F)

resultado

In [ ]:
p_valor <- resultado$p.value

p_valor

In [ ]:
significancia <- 0.05 

In [ ]:
if (p_valor <= significancia){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# Considerando o conjunto de informações abaixo:

In [ ]:
dados <- data.frame(
    Dados = c(116.48, -18.85, 150.38, 151.34, 105.08, 106.8, 0.12, 
        37.2, 152.73, 96.41, 111.03, 70.4, 98.05, 110.94, 33.61, 57.07, 
        123.72, 81.82, 105.77, 36.24, 79.45, 137.87, 85.7, 170.41, 
        68.46, 88.15, 60.26, 84.47, 168.2, 133.59, 84.49, 49.8, 104.52, 
        172.91, 121.5, 99.98, 208.15, 144.28, 105.39, 135.1)
)

# Aplique o teste de normalidade e selecione a alternativa correta. Considere 5% de significância.

In [ ]:
resultado <- shapiro.test(dados$Dados)

resultado

In [ ]:
p_valor <- resultado$p.value 

p_valor

In [ ]:
significancia <- 0.05

In [ ]:
if (p_valor <= significancia){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

# Você é um pesquisador que estuda o mercado de trabalho brasileiro e resolve estudar as diferenças salariais dos trabalhadores dos estados do Rio de Janeiro e de São Paulo. Durante sua pesquisa você verifica que, aparentemente, os rendimentos dos trabalhadores no estado do Rio de Janeiro são mais baixos que os rendimentos dos trabalhadores no estado de São Paulo. Para confirmar esta conclusão realize um teste de hipótese de comparação de médias em cima de duas amostras de trabalhadores dos dois estados. Siga as seguintes etapas:

- Utilize o *dataset* 'dados_proj.csv' para rodar o teste. Este *dataset* tem uma amostra de **tamanho 500** dos rendimentos dos trabalhadores dos dois estados (RJ e SP).
- Considere o **nível de significância de 5%**.
- Teste a hipótese de que a **renda média dos trabalhadores do Rio de Janeiro é menor que a renda média dos trabalhadores de São Paulo**.

In [10]:
dados <- read.csv("../input/dados-proj-3-parte-curso-r/dados_proj.csv")

In [11]:
head(dados)

Renda_RJ Renda_SP
1  1300     1800   
2 35000      700   
3  1300     8000   
4  2364     1800   
5  1500     2364   
6  2500    11000

**Obtendo média e desvio padrão para as duas amostras**

In [12]:
media_rj <- mean(dados$Renda_RJ)

media_rj

[1] 2240.518

In [13]:
sd_rj <- sd(dados$Renda_RJ)

sd_rj

[1] 4643.836

In [14]:
media_sp <- mean(dados$Renda_SP)

media_sp

[1] 2839.352

In [45]:
sd_sp <- sd(dados$Renda_SP)

sd_sp

[1] 3699.384

In [28]:
D_0 <- 0

In [29]:
n <- nrow(dados)

n

[1] 500

**Passo 1 - formulação das hipóteses**

*H0: u_rj >= u_sp*

*H1: u_rj < u_sp*

*ou*

*H0: u_rj - u_sp >= 0*

*H1: u_rj - u_sp < 0*

**Passo 2 - escolha da distruição amostral adequada**

In [30]:
nrow(dados)

[1] 500

*N > 30. Utilizo distribuição normal. Desvio padrao populacional desconhecido. Utilizo dsitribuicao da amostra*

**Passo 3 - fixação da significância do teste**

In [31]:
significancia <- 0.05

In [38]:
z_alpha <- qnorm(significancia)

z_alpha

[1] -1.644854

**Passo 4 - Cálculo da estátistica-teste e verificação desse valor com as áreas de aceitação e rejeição do teste**

In [39]:
numerador <- media_rj - media_sp - D_0

numerador

[1] -598.834

In [40]:
denominador <- sqrt((sd_rj ** 2 / n) + (sd_sp ** 2 / n))

denominador

[1] 265.5208

In [41]:
z <- numerador / denominador

z

[1] -2.255318

**Passo 5 - Aceitação ou rejeição da hipótese nula**

In [43]:
if (z <= z_alpha){    
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

[1] "Rejeitar H0"

# Uma empresa de consultoria alega que seus consultores recebem um salário médio igual a, no mínimo, reais 2.500,00, com desvio padrão de reais 500,00 e distribuição aproximadamente normal. Uma amostra aleatória de 300 consultores mostrou uma média de reais 2.400,00. A afirmação da empresa pode ser aceita a um nível de significância de 3%?

In [46]:
media <- 2500

In [47]:
sd <- 500

In [48]:
n <- 300

In [49]:
media_amostra <- 2400

In [50]:
significancia <- 0.03

In [51]:
z_alpha <- qnorm(significancia)

z_alpha

[1] -1.880794

In [52]:
z <- (media_amostra - media) / (sd / sqrt(n))

z

[1] -3.464102

In [53]:
if (z <= z_alpha){
    
    "Rejeitar H0"
    
} else {
    
    "Aceitar H0"
    
}

[1] "Rejeitar H0"